### PR TITLE
Update trusted application tutorial

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 1]
```diff
@@ -4,307 +4,216 @@
...
+touch protect-fs.sh
 ```
 
-{% hint style="warning" %}
-The file **utils/signer.py** is just a temporary workaround and it will be removed in the next release. But, for now, it is mandatory that's why we copy it inside the Dockerfile. It is called during the execution time by the worker.
-{% endhint %}
+In the folder `src/` create the file `app.js` \(or `app.py` if you want to use Python\) then copy [this](../your-first-app.md#write-the-app-shell-script-example) code inside.
 
-The `Dockerfile` is a ready-to-go template where you just need to add your system packages and application's dependencies in the dedicated block.
+As we mentioned earlier, the advantage of using **SCONE** is the ability to make the application **Intel SGX-enabled** without changing the source code. The only thing we are going to modify is the `Dockerfile`. First, we need to change the base image from the official `node (or python)` to the one provided by SCONE: `sconecuratedimages/apps:node-8.9.4-alpine-scone3.0`or `sconecuratedimages/apps:python-3.7.3-alpine3.10-scone3.0`. Those base docker images contain a `nodejs/python` interpreters that run inside enclaves.
```
Comment (Amxx) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]: Intel® SGX

Comment (Amxx) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]: `node` \(or `python`\)

Comment (Amxx) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]: `nodejs`/`python`

File: for-developers/confidential-computing/sgx-encrypted-dataset.md [hunk 2]
```diff
@@ -142,41 +137,253 @@ Check it by doing:
...
+const fs = require('fs');
+var figlet = require('figlet');
+
+const iexec_out = process.env.IEXEC_OUT;
+const iexec_in = process.env.IEXEC_IN;
+
+var text = "";
+
+// Eventually use some confidential assets
+if (fs.existsSync(iexec_in + '/dataset.txt')) {
```
Comment (PierreJeanjacquot) [inline for-developers/confidential-computing/sgx-encrypted-dataset.md]: dataset file name is accessible through env `IEXEC_DATASET_FILENAME`

File: for-developers/confidential-computing/sgx-encrypted-dataset.md [hunk 3]
```diff
@@ -142,41 +137,253 @@ Check it by doing:
...
-def write_file(path, data):
-    with open(path, "w+") as f:
-        f.write(data)
+iexec_out = os.environ['IEXEC_OUT']
+iexec_in = os.environ['IEXEC_IN']
+
+text = ""
+
+# Eventually use some confidential assets
+if os.path.exists(iexec_in + '/dataset.txt'):
```
Comment (PierreJeanjacquot) [inline for-developers/confidential-computing/sgx-encrypted-dataset.md]: same here

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
use `IEXEC_DATASET_FILENAME`

File: for-developers/confidential-computing/create-your-first-sgx-app.md (hunk 1 above)
Comment (zguesmi) [reply for-developers/confidential-computing/create-your-first-sgx-app.md]: Done, thanks!

File: for-developers/confidential-computing/sgx-encrypted-dataset.md (hunk 2 above)
Comment (zguesmi) [reply for-developers/confidential-computing/sgx-encrypted-dataset.md]: Thanks :+1:

**Review by james-toussaint — APPROVED** [review]

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**